extensions/amp-animation/0.1/runners/scrolltimeline-worklet.js

```diff
@@ -0,0 +1,192 @@
+    this.topRatio_ = viewportData['top-ratio'];
+
+    /** @private {number} */
+    this.bottomRatio_ = viewportData['bottom-ratio'];
+
+    /** @private {number} */
+    this.topMargin_ = viewportData['top-margin'];
+
+    /** @private {number} */
+    this.bottomMargin_ =
+      viewportData['bottom-margin'];
+  }
+
+  /**
+  * @override
+  * Initializes the players but does not change the state.
+   */
+  init() {
+    this.requests_.map(request => {
+      // Apply vars.
+      if (request.vars) {
+        setStyles(request.target,
+            assertDoesNotContainDisplay(request.vars));
+      }
+      // TODO(nainar): This switches all animations to AnimationWorklet.
+      // Limit only to Scroll based animations for now.
+      getOrAddWorkletModule(this.win_).then(() => {
+        const {documentElement} = this.win_.document;
+        const viewportService = Services.viewportForDoc(documentElement);
+
+        const scrollSource = viewportService.getScrollingElement();
+        const elementRect = request.target./*OK*/getBoundingClientRect();
+        const scrollTimeline = new this.win_.ScrollTimeline({
+          scrollSource,
+          orientation: 'block',
+          timeRange: request.timing.duration,
+          startScrollOffset: `${this.topMargin_}px`,
+          endScrollOffset: `${this.bottomMargin_}px`,
+          fill: request.timing.fill,
+        });
+        const keyframeEffect = new KeyframeEffect(request.target,
+            request.keyframes, request.timing);
+        const player = new this.win_.WorkletAnimation(`${moduleName}`,
+            [keyframeEffect],
+            scrollTimeline, {
+              'time-range': request.timing.duration,
+              'start-offset': this.topMargin_,
+              'end-offset': this.bottomMargin_,
+              'top-ratio': this.topRatio_,
+              'bottom-ratio': this.bottomRatio_,
+              'element-height': elementRect.height,
+            });
+        player.play();
+        this.players_.push(player);
+      }, e => {
+        dev().error('AMP-ANIMATION SCROLLTIMELINE-WORKLET', e);
+      });
+    });
+  }
+
+  /**
+   * @override
+   * Initializes the players if not already initialized,
+   * and starts playing the animations.
+   */
+  start() {
+    if (!this.players_) {
+      this.init();
+    }
+  }
+
+  /**
+   * @override
+   */
+  cancel() {
+    if (!this.players_) {
+      return;
+    }
+    this.players_.forEach(player => {
+      player.cancel();
+    });
+  }
+
+}
+
+/**
+ * @param {!Window} win
+ * @private
+ */
+function getOrAddWorkletModule(win) {
+  if (workletModulePromise) {
+    return workletModulePromise;
+  }
+  const blob =
+ `registerAnimator('${moduleName}', class {
+    constructor(options = {
+      'time-range': 0,
+      'start-offset': 0,
+      'end-offset': 0,
+      'top-ratio': 0,
+      'bottom-ratio': 0,
+      'element-height': 0,
+    }) {
+      console.info('Using animationWorklet ScrollTimeline');
```

Comment:
no changes in the code itself other than `registerAnimator('${moduleName}'` and `console.info('Using animationWorklet ScrollTimeline');`